enforcer-rules/README.md

```diff
@@ -3,6 +3,31 @@
 This Maven enforcer rule verifies that the transitive dependency tree of `pom.xml` does not have
 any [linkage errors](../library-best-practices/glossary.md#types-of-conflicts-and-compatibility).
 
+This rule performs the following steps:
+
+1. Build a dependency tree of Maven artifacts with the project as the root node
+
+   In the tree, a Maven artifact has directed edges to other Maven artifacts when the artifact's
+   pom.xml declares dependencies to them. The root node of the tree is the Maven artifact
+   of the project in which the enforce rule runs.
+
+   **`Optional` and `provided`**: Unlike Maven's resolution algorithm, the tree includes transitive
+   dependencies marked as `optional` or `provided`.
+   This inclusion helps to detect incompatibilities beyond Maven's build class path.
+1. Build a class path from the dependency tree
+
+   It traverses the tree in level-order to generate a list of Maven artifacts.
+1. Run [Linkage Checker](../dependencies) with the class path (list of JAR files) as input
```

Comment:
we don't want to reference linkage checker here. Just describe what this tool does.

Reply:
Removed linkage checker link.